lib/exceptions.h

```diff
@@ -48,13 +48,31 @@ class CompilerBug final : public P4CExceptionBase {
     template <typename... T>
     CompilerBug(const char* format, T... args)
             : P4CExceptionBase(format, args...)
-    { message = "COMPILER BUG:\n" + message; }
+    // \e[31m prints the text in red
+    { message = "\e[31mCOMPILER BUG\e[0m:\n" + message; }
     template <typename... T>
     CompilerBug(const char* file, int line, const char* format, T... args)
             : P4CExceptionBase(format, args...)
-    { message = cstring("COMPILER BUG: ") + file + ":" + Util::toString(line) + "\n" + message; }
+    { message = cstring("In file: ") + file + ":" + Util::toString(line) + "\n" +
+                cstring("\e[31mCOMPILER BUG\e[0m: ") + message; }
 };
 
+// This class indicates an unimplemented feature in the compiler
+class CompilerUnimplemented final : public P4CExceptionBase {
```

Comment:
It is definitely portable. Works even on mobile phones. Emacs shell as well: 
<img width="567" alt="screen shot 2017-05-02 at 17 53 44" src="https://cloud.githubusercontent.com/assets/3770874/25645243/04d057aa-2f61-11e7-9cc2-d4a2b3400048.png">